### PR TITLE
Simple interrupt example using kick-ass.

### PR DESCRIPTION
**Pull request by m00dawg**
This just types a string on the screen using VBLANK plus a speed counter but (hopefully) provides a nice example of how to make interrupts work as well as using some kernel (kernal?) routines.

All the other examples use acme, but I use kick-ass so I thought it might be nice to have a kick-ass directory for it (although the code can likely be converted without much fuss).

### CODE REVIEW

**Comment by rsbohn** [comment]
You need to use RTI instead of JMP LOOP in your interrupt handler, otherwise you're leaving stuff on the stack and it will quickly overflow.

**Comment by m00dawg** [comment]
Thanks very much for the info! That's super helpful! I looked at memory address $0314-0315 and yep it's EB01. So just to be clear, I can/should store whatever value is there when I setup my program so I can jump to whatever address that is? Or should I just assume EB01?

That routine would eventually then issue RTI?

**Comment by greg-king5** [comment]
> So, just to be clear, I can/should store whatever value is there when I set up my program, so I can jump to whatever address that is? Or, should I just assume $EB01?
> 
> That routine would eventually then issue RTI?

You should assume nothing about where things are inside the ROMs.  That code might be re-arranged.  Also, the Kernal resets VERA's interrupt after it has finished doing its work.  Thirdly, other pieces of code might have attached themselves to the interrupts (it's called "daisy-chaining").

Therefore, you must save the current vector before you replace it; and, you must jump through that vector after your handler is finished.  Also, don't reset VERA's interrupt -- other code needs to know that it happened.

**Comment by m00dawg** [comment]
Hmm interesting. Took me a bit to figure out I needed to use indirect addressing to make it work, but changes applied!

File: assembly/kick-assembler/typing.asm [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  /* If A is not 0, we know we do not need to reset the string so we
+     jump to the end since we are done. Otherwise, we reset the string
+     postion back to 0 so we can start printing the string all over again. */
+  cmp #0                // Compare A to end of string
+  bne typing_irq_end    // If it's not 0, go to end of handler
+  ldx #$00              // Otherwise reset string position
+  stx STRING_POSITION
+
+typing_irq_end:
+  /* Jump to the previous ISR handler. Not the use of indirect addressing. */
```
Comment (greg-king5) [inline assembly/kick-assembler/typing.asm]: Typo: `Not` -> `Note`

Comment (m00dawg) [reply assembly/kick-assembler/typing.asm]: Ah good call. Found a few more so I fixed those.

**Comment by m00dawg** [comment]
@greg-king5 
Since we're chatting, I'm trying to figure out a bit more about VBLANK's IRQ. You mentioned we shouldn't be clearing it since the kernal will as it needs to know about VBLANK. That works for this program, but doesn't for a similar program I've been writing (it does the same thing as this one but starts to insert random characters as Jack slowly goes insane). 

In that one, if I don't clear the VBLANK IRQ, it goes into an infitie loop (it just prints the 'A' in ALL WORK ....). If I clear the VBLANK, it works as expected and I can't quite figure out why. I'm wondering if it has to do with using JMPs (typing.asm doesn't do that until it jumps back to the previous ISR)? I have to jump over a some things within my subroutine (e.g. don't start printing random characters until X lines, and when we do, slowly increment the number of random characters allowed per line). Any thoughts?

**Pull request merged**